main.hs

```diff
@@ -175,354 +449,696 @@ readOodiNames = do
+          -> Config         -- ^ Argument: Pointer to the /config.yaml/ data.
+          -> [Maybe Table]  -- ^ Return:   List of 'Maybe' 'Table's 
 findTable c cnf = map ($| processTable cnf) (c $.// attributeIs "class" "confluenceTable" :: [Cursor])
 
-getHeader :: Cursor -> Maybe Header
-getHeader c = return x <* guard (not $ T.null x)
-  where x = T.toLower . normalize $ T.unwords (c $// content)
 
-processTable :: Config -> Cursor -> Maybe Table
+{- | This function takes a pointer to the 'Config' data and an XML 'Cursor'
+    (pointing to the class attributes with the value @confluenceTable@). It
+    picks out all the XML-elements beginning with the tag @tr@ from the
+    @confluenceTable@ class 'Cursor' and maps all child elements of the @tr@
+    element to it.
+
+    The @cells@ variable contains all the rows of the table. The first row
+    contains some higher-level 'Header's (eg. @Vastaava opettaja@), hence this
+    row is ignored. The second row contains the main 'Header's of the different
+    columns, and the rest of the rows are either 'Course' information or
+    'Category's separating the different 'Course' informations.
+
+    The first column in the main 'Header' row is empty (this is the column
+    containing the 'Category' headers when looking at the Wiki Table),
+    therefore only the 'tail' of it is necessary.
+-}
+processTable :: Config      -- ^ Argument: Pointer to the /config.yaml/ file.
+             -> Cursor      -- ^ Argument: XML document 'Cursor' pointing at @confluenceTable@ /class/ attribute.
+             -> Maybe Table -- ^ Return: The processed 'Maybe' 'Table'.
 processTable cnf c = case cells of
-    _ : header : xs ->
-        let headers       = mapMaybe getHeader header
+    _ : headersRaw : xs ->
+        let headers       = tail (mapMaybe getHeader headersRaw)
             (_, mcourses) = L.mapAccumL (getRow cnf headers) [] xs
         in Just $ Table (unsafePerformIO getCurrentTime) headers (catMaybes mcourses)
     _               -> Nothing
   where
     cells = map ($/ anyElement) (c $// element "tr")
 
--- | A row is either a category or course. The @[Category]@ is used as an
--- accumulator.
-getRow :: Config -> [Header] -> [Category] -> [Cursor] -> ([Category], Maybe Course)
-getRow cnf@Config{..} hs cats cs = map (T.unwords . ($// content)) cs `go` head (cs !! 1 $| attribute "class")
-    where go []        _       = (cats, Nothing)
-          go (mc : vs) classes = case toCategory cnf mc of
-                Just cat                        -> (accumCategory cnf cat cats, Nothing)
-                Nothing | null vs               -> (cats, Nothing)
-                        | T.null (normalize mc) -> (cats, Just $ toCourse cnf cats hs (classCur `T.isInfixOf` classes) vs)
-                        | otherwise             -> (cats, Just $ toCourse cnf cats hs (classCur `T.isInfixOf` classes) vs)
+
+{- | Create a 'Maybe' 'Header' type corresponding to the value of the raw
```

Comment:
Create a Maybe Header type corresponding to the value of type Maybe Header?
Create a value of type Maybe Header corresponding to the value of the raw?